plugins/destination/file/client/spec.go

```diff
@@ -32,8 +32,6 @@ func (s *Spec) SetDefaults() {
 		if !s.NoRotate {
 			s.Path += "." + PathVarUUID
 		}
-	} else {
-		s.Directory = path.Dir(s.Path)
```

Comment:
`s.Directory` is now not used in the processing logic, it's only used for deriving `Path` and printing a warning.